Styling/Feat: UI and UX Enhancements in MainView, MovieCard, SignupView, and LoginView Components

MaGloCon opened the pull request:
Made several updates to improve the user interface and user experience of the MainView, MovieCard, SignupView, and LoginView components. 

Changes include:

1. **Styling**
  - Refactored and added code to use React-Bootstrap and Bootstrap
  - Added SCSS files for components that required custom styling 
  
2. *Assets*
   - Added an Assets folder to src repository with favicon and logo files
   - Added favicon to index.html.
   
3. **MainView Component**: 
   - Implemented a loading spinner.
   - Added pagination logic. (will likely be moved to API)
   - Improved layout and styling.

4. **MovieCard Component**: 
   - Added FontAwesomeIcon to be turned into a button, so that user can add movie to favorites.
   - Updated styling and layout (minor)

6. **SignupView and LoginView Components**: 
   - Refactored components to use Bootstrap for better UI.
   - Imported and used logo.
   - Added 'Already have an account?' and 'Don't have an account?' links to switch between Signup and Login views.

MaGloCon committed:
feat: Enhanced MovieCard component and updated MainView

- Added several enhancements to the MovieCard component, including the addition of a new movie-card.scss file for styling.
- MovieCard component now includes a Card from react-bootstrap, and the movie's title, director, and year are now displayed on the card.

- Updated the MainView component to import and use the Container, Navbar, Nav, Col, and Row components from react-bootstrap. The movies are now displayed in a grid layout using these components.

- Updated PropTypes for the MovieCard component to reflect the new properties of the movie objects.

MaGloCon committed:
feat: enhanced movie card styling/interactivity, updatedMainView

- Updated movie-card.scss to improve card styling, add hover effects, and adjust layout.
- Modified movie-card.jsx to include hover state, display additional movie information on hover, and restructure component for improved interactivity.
- Updated MainView:
  - Refactored imports from 'react-bootstrap'.
  - Added pagination to the movie list.
  - Implemented logic to display a subset of movies based on the current page.

MaGloCon committed:
feat: enhanced MainView and MovieCard components

- Consolidated react-bootstrap imports in MainView
- Implemented loading spinner in MainView
- Added pagination logic in MainView
- Improved layout and styling in MainView
- Added FontAwesomeIcon in MovieCard
- Updated styling and layout in MovieCard
- Added hover effect in MovieCard

MaGloCon committed:
Styled and Refactored SignupView, LoginView, MainView, MovieCard and MovieView

feat: Enhance UI and UX of Signup and Login views

- Added favicon to index.html
- Refactored SignupView and LoginView components to use Bootstrap for better UI
- Imported and used logo in SignupView and LoginView
- Added SCSS files for SignupView and LoginView for custom styling
- Add 'Already have an account?' and 'Don't have an account?' links to switch between Signup and Login views

Stanley-Okwii commented on package.json:
```diff
@@ -2,14 +2,20 @@
   "version": "1.0.0",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
```
Consider removing the `test` script here because it's something that you don't use. It was merely a template example.

Stanley-Okwii commented on src/components/login-view/login-view.jsx:
```diff
@@ -35,28 +40,41 @@ export const LoginView = ({ onLoggedIn }) => {
+            <Form.Label className="fs-6">Password <span className="text-danger">*</span></Form.Label>
+              <Form.Control
+                type="password"
+                // placeholder="Enter password"
```
A final submission should not include commented out code. It only creates clutter that you want to avoid. The same for all commented lines of code in this file.

Stanley-Okwii commented on src/components/main-view/main-view.jsx:
```diff
@@ -1,21 +1,35 @@
+
+  const [movies, setMovies] = useState([]);
+  const [selectedMovie, setSelectedMovie] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
```
The default loading state should always be `false`. It's only set to true, just before the API call is made.

Stanley-Okwii commented on src/components/main-view/main-view.jsx:
```diff
@@ -1,21 +1,35 @@
+  useEffect(() => {
     if (!token) return;
 
+    setIsLoading(true); 
```
This is the right correct place to set loading to `true`. 🎉

Stanley-Okwii commented on src/components/main-view/main-view.jsx:
```diff
@@ -38,54 +52,103 @@ export const MainView = () => {
         }
       })
-      .catch(error => console.error('Error:', error));
+      .catch(error => console.error('Error:', error))
```
After this line, you also want to call `setIsLoading(false);` because the loading spinner is constantly be displayed if an error occurred and the loading state isn't set to false.

MaGloCon replied:
Can I resolve this by placing .finally(() => setIsLoading(false))? after the .catch method?

Stanley-Okwii replied:
> Can I resolve this by placing .finally(() => setIsLoading(false))? after the .catch method?

Yeah, that's perfect.

Stanley-Okwii commented on src/components/movie-card/movie-card.jsx:
```diff
@@ -1,18 +1,49 @@
+        <Card.Body className="hovered-body bg-white shadow pt-1">
+          <div className="d-flex justify-content-between">
+            <Button variant="btn btn-primary" onClick={() => {onMovieClick(movie)}}>More</Button> 
+            {/* <div onClick={yourClickHandlerFunction}> */}
```
Consider removing the commented out lines of code to declutter your code.

Stanley-Okwii commented on src/components/main-view/main-view.jsx:
```diff
@@ -38,54 +52,103 @@ export const MainView = () => {
+            </Row>
+          </Container>
+        ) : (
+          <Container className="d-flex justify-content-center align-items-center vh-100">
+            <Row className="d-flex justify-content-center align-items-center vw-100">
+              <Col sm={12} md={8}>
```
No action required.  As a rule of thumb, whenever you see yourself writing similar code, always extract it to a function or component that can be reused across multiple components.

Stanley-Okwii commented on dist/index.html:
```diff
@@ -1,7 +1,10 @@
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <title>Cinephile</title>
+    <link rel="icon" href="/logo.138479e9.ico" type="image/x-icon">
```
Nice touch 🎉

Stanley-Okwii reviewed (approved):
Great job  🚀, 
I've left a couple points for improvement.

MaGloCon committed:
refactor: Cleaned up and refactored state initialization

- Removed commented out code
- Refactored state initialization: Changed the initial value of isLoading state from true to false to correctly reflect the initial non-loading state. Added a finally block to the fetch operation to ensure isLoading is set to false even when an error occurs.
